src/wl_init.c

```diff
@@ -1078,7 +1081,10 @@ int _glfwPlatformInit(void)
 
     if (_glfw.wl.pointer && _glfw.wl.shm)
     {
-        _glfw.wl.cursorTheme = wl_cursor_theme_load(NULL, 32, _glfw.wl.shm);
+        cursorTheme = getenv("XCURSOR_THEME");
+        cursorSize = getenv("XCURSOR_SIZE") ? atoi(getenv("XCURSOR_SIZE")) : 32;
+        _glfw.wl.cursorTheme =
+            wl_cursor_theme_load(cursorTheme, cursorSize, _glfw.wl.shm);
```

Comment:
There is another `wl_cursor_theme_load()` a few lines below, you can use `cursorSize * 2` there for the size, as we only support outputs with `scale=1` and `scale=2`.